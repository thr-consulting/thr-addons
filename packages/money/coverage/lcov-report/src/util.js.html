<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/util.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">src</a> util.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.59% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>25/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/16</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Money from 'js-money';
import accounting from 'accounting';
import transform from 'lodash/transform';
import isNumber from 'lodash/isNumber';
import isArray from 'lodash/isArray';
import isObject from 'lodash/isObject';
import map from 'lodash/map';
import isString from 'lodash/isString';
import has from 'lodash/has';
&nbsp;
/**
 * Creates a Money object out of a plain Object or decimal. Defaults null to $0 CAD.
 * @param {Object|number|string} objOrDecimal - The object or decimal to convert to Money.
 * @param {string} currency - If converting a decimal, null, or string, specifies the currency to use.
 * @return {Money}
 */
export function makeMoney(objOrDecimal, currency = Money.CAD) {
	if (objOrDecimal instanceof Money) return objOrDecimal;
	if (!objOrDecimal) return new Money(0, currency);
	if (isNumber(objOrDecimal)) return Money.fromDecimal(objOrDecimal, currency, 'round');
	if (isObject(objOrDecimal) &amp;&amp; has(objOrDecimal, 'amount') &amp;&amp; has(objOrDecimal, 'currency')) return new Money(objOrDecimal.amount, objOrDecimal.currency);
	if (isString(objOrDecimal)) return Money.fromDecimal(parseFloat(objOrDecimal), currency, 'round');
	throw new Error(`Can't convert value to Money: ${objOrDecimal}.`);
}
&nbsp;
/**
 * Rounds a decimal value to a certain number of decimal digits.
 * @param {number} value - A number to round.
 * @param {number} decimals - The number of decimal digits to round to.
 * @return {number} - The rounded number.
 */
export function roundTo(value, decimals) {
	return Number(`${Math.round(`${value}e${decimals}`).toString()}e-${decimals}`);
}
&nbsp;
/**
 * Formats a Money value to a nice string.
 * @param {Money} money - The Money value
 * @param {bool} symbol - If true displays the proper currency symbol
 * @return {string} The formatted Money string
 */
export function formatMoney(money, symbol = false) {
	<span class="missing-if-branch" title="if path not taken" >I</span>if (!money) <span class="cstat-no" title="statement not covered" >return '';</span>
	return accounting.formatMoney(money.toDecimal(), symbol ? Money[money.currency].symbol_native : '', Money[money.currency].decimal_digits);
}
&nbsp;
/**
 * Transforms anything that has Money style objects into Money.
 * @param obj
 * @returns {*}
 */
export function transformObjectsToMoney(obj) {
	if (isArray(obj)) {
		return map(obj, v =&gt; transformObjectsToMoney(v));
	}
	<span class="missing-if-branch" title="else path not taken" >E</span>if (isObject(obj)) {
		if (obj.amount &amp;&amp; obj.currency) return new Money(obj.amount, obj.currency);
		return transform(obj, (result, value, key) =&gt; {
			result[key] = transformObjectsToMoney(value); // eslint-disable-line no-param-reassign
		}, {});
	}
<span class="cstat-no" title="statement not covered" >	return obj;</span>
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jun 21 2018 15:55:42 GMT-0500 (CDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
<script src="../block-navigation.js"></script>
</body>
</html>
