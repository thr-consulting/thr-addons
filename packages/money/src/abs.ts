import Money from 'js-money';

export const abs = (money: Money) => new Money(Math.abs(money.getAmount()), money.currency);
