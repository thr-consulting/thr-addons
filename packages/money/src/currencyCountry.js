// @flow

const currencyCountryMap = {
	USD: 'US',
	CAD: 'CA',
	EUR: 'DE',
	AED: 'AE',
	AFN: 'AF',
	ALL: 'AL',
	AMD: 'AM',
	ARS: 'AR',
	AUD: 'AU',
	AZN: 'AZ',
	BAM: 'BA',
	BDT: 'BD',
	BGN: 'BG',
	BHD: 'BH',
	BIF: 'BI',
	BND: 'BN',
	BOB: 'BO',
	BRL: 'BR',
	BWP: 'BW',
	BYR: 'BY',
	BZD: 'BZ',
	CDF: 'CD',
	CHF: 'CH',
	CLP: 'CL',
	CNY: 'CN',
	COP: 'CO',
	CRC: 'CR',
	CVE: 'CV',
	CZK: 'CZ',
	DJF: 'DJ',
	DKK: 'DK',
	DOP: 'DO',
	DZD: 'DZ',
	EEK: 'EE',
	EGP: 'EG',
	ERN: 'ER',
	ETB: 'ET',
	GBP: 'GB',
	GEL: 'GE',
	GHS: 'GH',
	GNF: 'GN',
	GTQ: 'GT',
	HKD: 'HK',
	HNL: 'HN',
	HRK: 'HR',
	HUF: 'HU',
	IDR: 'ID',
	ILS: 'IL',
	INR: 'IN',
	IQD: 'IQ',
	IRR: 'IR',
	ISK: 'IS',
	JMD: 'JM',
	JOD: 'JO',
	JPY: 'JP',
	KES: 'KE',
	KHR: 'KH',
	KMF: 'KM',
	KRW: 'KR',
	KWD: 'KW',
	KZT: 'KZ',
	LBP: 'LB',
	LKR: 'LK',
	LTL: 'LT',
	LVL: 'LV',
	LYD: 'LY',
	MAD: 'MA',
	MDL: 'MD',
	MGA: 'MG',
	MKD: 'MK',
	MMK: 'MM',
	MOP: 'MO',
	MUR: 'MU',
	MXN: 'MX',
	MYR: 'MY',
	MZN: 'MZ',
	NAD: 'NA',
	NGN: 'NG',
	NIO: 'NI',
	NOK: 'NO',
	NPR: 'NP',
	NZD: 'NZ',
	OMR: 'OM',
	PAB: 'PA',
	PEN: 'PE',
	PHP: 'PH',
	PKR: 'PK',
	PLN: 'PL',
	PYG: 'PY',
	QAR: 'QA',
	RON: 'RO',
	RSD: 'RS',
	RUB: 'RU',
	RWF: 'RW',
	SAR: 'SA',
	SDG: 'SD',
	SEK: 'SE',
	SGD: 'SG',
	SOS: 'SO',
	SYP: 'SY',
	THB: 'TH',
	TND: 'TN',
	TOP: 'TO',
	TRY: 'TR',
	TTD: 'TT',
	TWD: 'TW',
	TZS: 'TZ',
	UAH: 'UA',
	UGX: 'UG',
	UYU: 'UY',
	UZS: 'UZ',
	VEF: 'VE',
	VND: 'VN',
	XAF: 'CM',
	XOF: 'CF',
	YER: 'YE',
	ZAR: 'ZA',
	ZMK: 'ZM',
};

/**
 * Returns a country code from a currency. Not 100% accurate as some currencies are used in multiple countries.
 * @param {string} currency - The currency code.
 * @return {string} The country code.
 */
export default function getCountryCode(currency: string) {
	if (!currencyCountryMap[currency]) return '';
	return currencyCountryMap[currency].toLowerCase();
}
