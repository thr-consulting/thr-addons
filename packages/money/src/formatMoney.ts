import accounting from 'accounting';
import Money from 'js-money';

/**
 * Formats a Money value to a nice string.
 * @param {Money} value - The Money value
 * @param {boolean} symbol - If true displays the proper currency symbol
 * @return {string} The formatted Money string
 */
export function formatMoney(value: Money | undefined | null = null, symbol = false): string {
	if (!value) return '';
	return accounting.formatMoney(
		value.toDecimal(),
		// @ts-ignore
		symbol ? Money[value.currency].symbol_native : '',
		// @ts-ignore
		Money[value.currency].decimal_digits,
	);
}
