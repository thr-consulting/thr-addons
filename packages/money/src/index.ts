// @ts-ignore
export {default as Money} from 'js-money';
export {default as Currencies} from 'js-money/lib/currency';
export {CurrencyCountry} from './currencyCountry';
export {toMoney, isMoney, isMoneyObject, IMoneyObject} from './toMoney';
export {formatMoney} from './formatMoney';
export {roundTo} from './roundTo';
export {abs} from './abs';
