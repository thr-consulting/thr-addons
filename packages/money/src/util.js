import Money from 'js-money';
import accounting from 'accounting';
import transform from 'lodash/transform';
import isNumber from 'lodash/isNumber';
import isArray from 'lodash/isArray';
import isObject from 'lodash/isObject';
import map from 'lodash/map';

/**
 * Creates a Money object out of a plain Object or decimal. Defaults null to $0 CAD.
 * @param {Object|decimal} objOrDecimal - The object or decimal to convert to Money.
 * @param {string} currency - If converting a decimal, specifies the currency to use.
 * @return {Money}
 */
export function makeMoney(objOrDecimal, currency = 'CAD') {
	if (!objOrDecimal) return new Money(0, Money.CAD);
	if (isNumber(objOrDecimal)) return Money.fromDecimal(objOrDecimal, currency);
	if (objOrDecimal.amount && objOrDecimal.currency) return new Money(objOrDecimal.amount, objOrDecimal.currency);
	return new Money(0, Money.CAD);
}

/**
 * Rounds a decimal value to a certain number of decimal digits.
 * @param {number} value - A number to round.
 * @param {number} decimals - The number of decimal digits to round to.
 * @return {number} - The rounded number.
 */
export function roundTo(value, decimals) {
	return Number(`${Math.round(`${value}e${decimals}`).toString()}e-${decimals}`);
}

/**
 * Formats a Money value to a nice string.
 * @param {Money} money - The Money value
 * @param {bool} symbol - If true displays the proper currency symbol
 * @return {string} The formatted Money string
 */
export function formatMoney(money, symbol = false) {
	return accounting.formatMoney(money.toDecimal(), symbol ? Money[money.currency].symbol_native : '', Money[money.currency].decimal_digits);
}

/**
 * Transforms anything that has Money style objects into Money.
 * @param obj
 * @returns {*}
 */
export function transformObjectsToMoney(obj) {
	if (isArray(obj)) {
		return map(obj, v => transformObjectsToMoney(v));
	}
	if (isObject(obj)) {
		if (obj.amount && obj.currency) return new Money(obj.amount, obj.currency);
		return transform(obj, (result, value, key) => {
			result[key] = transformObjectsToMoney(value); // eslint-disable-line no-param-reassign
		}, {});
	}
	return obj;
}
