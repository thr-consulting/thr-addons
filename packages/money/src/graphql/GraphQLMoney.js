import {GraphQLError} from 'graphql/error';
import {GraphQLScalarType} from 'graphql/type';
import {Kind} from 'graphql/language';
import isString from 'lodash/isString';
import isObject from 'lodash/isObject';
import isNumber from 'lodash/isNumber';
import Money from 'js-money';
import {makeMoney} from '../util';

const GraphQLMoney = new GraphQLScalarType({
	name: 'Money',
	description: 'JS-Money object',
	// Parses variable values (JSON => JSON)
	parseValue(value) {
		return makeMoney(value);
	},
	// Called when the value is being sent to the client
	serialize(value) {
		if (value instanceof Money) return value;
		if (isObject(value) && value.amount && value.currency) return makeMoney(value);
		if (isString(value)) return makeMoney(value);
		if (isNumber(value)) return makeMoney(value);
		throw new GraphQLError('Trying to serialize a non-primitive');
	},
	// Parses GraphQL language AST into the value. (AST => JSON)
	parseLiteral(ast) {
		if (ast.kind === Kind.FLOAT || ast.kind === Kind.INT || ast.kind === Kind.STRING) return makeMoney(Number(ast.value));
		throw new GraphQLError(`Cannot convert literal ${ast.value} into Money.`);
	},
});

export default GraphQLMoney;
