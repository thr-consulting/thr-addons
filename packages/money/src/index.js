// @flow

import MoneyInput from './MoneyInput';
import getCountryCode from './currencyCountry';
import {makeMoney, formatMoney, roundTo, transformObjectsToMoney} from './util';

export {
	MoneyInput,
	getCountryCode,
	makeMoney,
	formatMoney,
	roundTo,
	transformObjectsToMoney,
};

/**
 * A JS-Money object.
 * @typedef Money
 */
