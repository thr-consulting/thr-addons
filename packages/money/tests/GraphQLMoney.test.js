import {graphql} from 'graphql';
import gql from 'graphql-tag';
import Money from 'js-money';
import {makeExecutableSchema} from 'graphql-tools';
import GraphQLMoney from '../src/graphql/GraphQLMoney';
import {makeMoney} from '../src/util';

expect.addSnapshotSerializer({
	test: v => Object.prototype.toString.call(v) === '[object Object]',
	print: v => JSON.stringify(v),
});

const typeDefs = gql`
	scalar Money
	type Query {
		money(item: Money): Money
	}
	schema {query: Query}
`;

const resolvers = {
	Money: GraphQLMoney,
	Query: {
		money: (root, {item}) => item,
	},
};
const schema = makeExecutableSchema({typeDefs, resolvers});

expect.addSnapshotSerializer({
	test: v => v instanceof Money,
	print: v => `${v.toString()}`,
});

describe('Money GraphQL', () => {
	it('should validate literal string value', async () => {
		expect(graphql(schema, '{money(item: "4.56")}')).resolves.toMatchSnapshot();
	});

	it('should validate literal float value', () => {
		expect(graphql(schema, '{money(item: 4.56)}')).resolves.toMatchSnapshot();
	});

	it('should validate literal int value', () => {
		expect(graphql(schema, '{money(item: 4)}')).resolves.toMatchSnapshot();
	});

	it('should validate variables of type Money', () => {
		const query = 'query basic($val: Money) {money(item: $val)}';
		expect(graphql(schema, query, null, null, {val: makeMoney(4.56)})).resolves.toMatchSnapshot();
	});

	it('should validate variables of type float', () => {
		const query = 'query basic($val: Money) {money(item: $val)}';
		expect(graphql(schema, query, null, null, {val: 4.56})).resolves.toMatchSnapshot();
	});

	it('should validate variables of type int', () => {
		const query = 'query basic($val: Money) {money(item: $val)}';
		expect(graphql(schema, query, null, null, {val: 4})).resolves.toMatchSnapshot();
	});

	it('should validate variables of type string', () => {
		const query = 'query basic($val: Money) {money(item: $val)}';
		expect(graphql(schema, query, null, null, {val: '4.56'})).resolves.toMatchSnapshot();
	});

	it('should fail when an invalid variable is passed', () => {
		const query = 'query basic($val: Money) {money(item: $val)}';
		expect(graphql(schema, query, null, null, {val: {something: 'else'}})).resolves.toMatchSnapshot();
	});
});
