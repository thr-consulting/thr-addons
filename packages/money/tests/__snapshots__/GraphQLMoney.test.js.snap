// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money GraphQL should fail when an invalid variable is passed 1`] = `{"errors":[{"message":"Variable \\"$val\\" got invalid value { something: \\"else\\" }; Expected type Money; Can't convert value to Money: [object Object].","locations":[{"line":1,"column":13}]}]}`;

exports[`Money GraphQL should validate literal float value 1`] = `{"data":{"money":{"amount":456,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate literal int value 1`] = `{"data":{"money":{"amount":400,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate literal string value 1`] = `{"data":{"money":{"amount":456,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate variables of type Money 1`] = `{"data":{"money":{"amount":456,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate variables of type float 1`] = `{"data":{"money":{"amount":456,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate variables of type int 1`] = `{"data":{"money":{"amount":400,"currency":"CAD"}}}`;

exports[`Money GraphQL should validate variables of type string 1`] = `{"data":{"money":{"amount":456,"currency":"CAD"}}}`;
